### PR TITLE
HumanEvalFix integration

### PR DESCRIPTION
**Pull request by Muennighoff**
Integrates HumanEvalFix from https://arxiv.org/abs/2308.07124

### CODE REVIEW

**Comment by li-boxuan** [comment]
@Muennighoff This is awesome, thank you! Is it ready to try out, or still under development?

**Comment by yufansong** [comment]
@Muennighoff If it is done, could you plz elaborate more context in  PR discription? If you are still dev, you can make this PR as a draft one.

File: evaluation/humanevalfix/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+Create a `config.toml` file if it does not exist at the root of the workspace.
+
+Add the following configurations:
+
+```toml
+[core]
+max_iterations = 100
+cache_dir = "/tmp/cache"
+ssh_hostname = "localhost"
```
Comment (li-boxuan) [inline evaluation/humanevalfix/README.md:24]: You probably also want to include `enable_auto_lint = true`. Evaluation of CodeActAgent on `SWE-bench-lite` shows that this option could give the LLM a hint of indentation errors, and thus boosts the final score.

**Comment by Muennighoff** [comment]
> @Muennighoff I see there are some todos in your doc. If it is done, could you plz elaborate more context in PR discription? If you are still dev, you can make this PR as a draft one.

I've converted to draft, sorry! One todo is regarding programming languages - I'm unsure if it makes sense to also add evaluation for the other prog. langs in HumanEvalFix (Rust, C++, Java, JS, Go) or only Python?

**Comment by Muennighoff** [comment]
Also cc @tangxiangru who is also working on the integration

**Comment by li-boxuan** [comment]
> I'm unsure if it makes sense to also add evaluation for the other prog. langs in HumanEvalFix (Rust, C++, Java, JS, Go) or only Python?

I haven't read your paper yet, so please take my thoughts as a grain of salt:
1. We should prioritize whatever is easier to do first.
2. It's a great idea to include other programming languages. SWE-bench only involves python repositories, which doesn't 100% reflect the quality of OpenDevin in a broader scope. In reality, people use OpenDevin not just for python tasks.

**Comment by Muennighoff** [comment]
> Overall looks great to me! Without too much efforts, i can actually run this and get some meaningful results!
> 
> image

Amazing thanks so much for taking a look & your fixes! I have moved the PR out of draft mode.
Tagging @huybery & @tangxiangru in case they want to take a look 🤗

File: evaluation/humanevalfix/run_infer.py [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+
+LANGUAGE_TO_NUM_WORKERS = {
+    'python': 4,
+}
+
+
+def cleanup():
+    print('Cleaning up child processes...')
+    for process in mp.active_children():
+        print(f'Terminating child process: {process.name}')
```
Comment (yufansong) [inline evaluation/humanevalfix/run_infer.py:68]:
```suggestion
    logger.info('Cleaning up child processes...')
    for process in mp.active_children():
        logger.info(f'Terminating child process: {process.name}')
```

**Review by yufansong — APPROVED** [review]
Leave some nits. Most LGTM

File: evaluation/humanevalfix/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+Implements evaluation of agents on HumanEvalFix from the HumanEvalPack benchmark introduced in [OctoPack: Instruction Tuning Code Large Language Models](https://arxiv.org/abs/2308.07124). Please see https://github.com/bigcode-project/bigcode-evaluation-harness/blob/main/bigcode_eval/tasks/humanevalpack.py for the reference implementation used in the paper.
+
+## Setup Environment
+
+Please follow [this document](https://github.com/OpenDevin/OpenDevin/blob/main/Development.md) to setup local develop environment for OpenDevin.
+
+In addition, evaluation requires the `evaluate` package installable via:
+```bash
+pip install evaluate
+```
```
[the remark this answers is not in the document]
Comment (tangxiangru) [reply evaluation/humanevalfix/README.md]: @yufansong fixed. now installing evaluate is in the dependency file.

**Pull request merged**